app/src/main/java/it/niedermann/nextcloud/deck/persistence/sync/SyncManager.java

```diff
@@ -166,76 +172,130 @@ public void onError(Throwable throwable) {
 
     @AnyThread
     public void synchronize(@NonNull IResponseCallback<Boolean> responseCallback) {
-        if(responseCallback.getAccount() == null) {
+        synchronize(Collections.singletonList(responseCallback));
+    }
+    private void synchronize(@NonNull List<IResponseCallback<Boolean>> responseCallbacks) {
+        IResponseCallback responseCallback = responseCallbacks.get(0);
```

Comment:
Please add a check or at least annotate the `responseCallbacks` argument with `@Size(min=1)`

Reply:
Wouldn't have happened with the current impl, but might get relevant later on, fixed.